axelrod/fingerprint.py

```diff
@@ -0,0 +1,285 @@
+            original player, the second is the dual, the rest are the probes.
+
+        """
+        probe_coordinates = create_coordinates(step)
+        self.coordinates = probe_coordinates
+        edges = create_edges(probe_coordinates)
+
+        dual = DualTransformer()(self.strategy)()
+        probe_players = create_probes(self.probe, probe_coordinates)
+        # probes = probe_players.values()
+        tournament_players = [self.strategy(), dual] + probe_players
+
+        return edges, tournament_players
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None):
+        """Build and play the spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        step : float, optional
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+        processes : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        outputfile = NamedTemporaryFile(mode='w')
+        filename = outputfile.name
+        edges, tourn_players = self.construct_tournament_elements(step)
+        self.spatial_tourn = axl.SpatialTournament(tourn_players,
+                                                   turns=turns,
+                                                   repetitions=repetitions,
+                                                   edges=edges)
+        self.spatial_tourn.play(processes=processes,
+                                build_results=False,
+                                filename=filename)
+        self.interactions = read_interactions(filename)
+        self.data = generate_data(self.interactions, self.coordinates, edges)
+
+    def plot(self, filename=None, col_map='seismic'):
+        """Plot the results of the spatial tournament.
+
+        Parameters
+        ----------
+        filename : str, optional
+            The location and name that the resulting plot should be saved to.
+            Defaults to the current directory with the name
+            `Strategy and Probe.pdf`
+        col_map : str, optional
+            A matplotlib colour map, full list can be found at
+            http://matplotlib.org/examples/color/colormaps_reference.html
+        """
+
+        # sns.heatmap(self.data, cmap=col_map)
+        # plt.show()
+        from math import sqrt
+        granularity = int(sqrt(len(self.coordinates)))
+        if filename is None:
```

Comment:
I see why you're doing it like this now though. Leave it as it is for now perhaps and let's see what it all looks like once everything else is in place (will need tests for edge cases to make sure something strange can't happen).